databricks/provider_test.go

```diff
@@ -76,3 +79,78 @@ resource "databricks_scim_group" "my-group-azure3" {
 }
 `
 }
+
+func TestProvider(t *testing.T) {
+	if err := testAccProvider.InternalValidate(); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+}
+
+func TestProvider_NoOptionsResultsInError(t *testing.T) {
+	var provider = Provider("")
+	var raw = make(map[string]interface{})
+	raw["config_file"] = "testdata/.databrickscfg_non_existant"
+	err := provider.Configure(terraform.NewResourceConfigRaw(raw))
+	assert.NotNil(t, err)
+}
+
+func TestProvider_HostTokensTakePrecedence(t *testing.T) {
+	var raw = make(map[string]interface{})
+	raw["host"] = "foo"
+	raw["token"] = "configured"
+	raw["config_file"] = "testdata/.databrickscfg"
+	err := testAccProvider.Configure(terraform.NewResourceConfigRaw(raw))
+	assert.Nil(t, err)
+
+	client := testAccProvider.Meta().(*service.DBApiClient).Config
+	assert.Equal(t, "configured", client.Token)
+}
+
+func TestProvider_MissingEnvMakesConfigRead(t *testing.T) {
+	var raw = make(map[string]interface{})
+	raw["token"] = "configured"
+	raw["config_file"] = "testdata/.databrickscfg"
+	err := testAccProvider.Configure(terraform.NewResourceConfigRaw(raw))
+	assert.Nil(t, err)
+
+	client := testAccProvider.Meta().(*service.DBApiClient).Config
+	assert.Equal(t, "PT0+IC9kZXYvdXJhbmRvbSA8PT0KYFZ", client.Token)
+}
+
+func TestProvider_NoHostGivesError(t *testing.T) {
+	var raw = make(map[string]interface{})
+	raw["config_file"] = "testdata/.databrickscfg"
+	raw["profile"] = "nohost"
+	err := testAccProvider.Configure(terraform.NewResourceConfigRaw(raw))
+	assert.NotNil(t, err)
+}
+
+func TestProvider_NoTokenGivesError(t *testing.T) {
+	var raw = make(map[string]interface{})
+	raw["config_file"] = "testdata/.databrickscfg"
+	raw["profile"] = "notoken"
+	err := testAccProvider.Configure(terraform.NewResourceConfigRaw(raw))
+	assert.NotNil(t, err)
+}
+
+func TestProvider_InvalidProfileGivesError(t *testing.T) {
+	var raw = make(map[string]interface{})
+	raw["config_file"] = "testdata/.databrickscfg"
+	raw["profile"] = "invalidhost"
+	err := testAccProvider.Configure(terraform.NewResourceConfigRaw(raw))
+	assert.NotNil(t, err)
```

Comment:
It might also be a good idea to actually check that the error message is the one that you expect and contains all the context required.

Reply:
according to certain beliefs, asserting exception message was not considered a good practice - e.g. because it may change the content of the string or be translated to other language, therefore making test "more overfitting" and fragile to insignificant changes.